### PR TITLE
 Required changes in agent simulator for integration tests #1105 

### PR DESCRIPTION
**Pull request by Pac0x73706f74686f756e64**
Closes https://github.com/wazuh/wazuh-qa/issues/1105

We have some commonly required changes on the agent simulator for our remoted integration tests that will be tracked on this issue.

- Add a buffer/queue for received messages accessible from the tests using the simulator.
- Allow merged.mg update on agent simulator to avoid having the Manager sending the shared files once and again.
- Small refactor to avoid issues with "tcp" and "TCP" different string in comparisons.
- Minor improvements in agent simulator to enhance test execution.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py [hunk 1]
```diff
@@ -891,23 +911,29 @@ class Sender:
...
             length = pack('<I', len(event))
-            self.socket.send(length + event)
-        if self.protocol == 'UDP':
+            try:
+                self.socket.send(length + event)
+            except BrokenPipeError:
+                logging.warning(f"Broken Pipe error while sending event. Creating new socket...")
+                sleep(5)
+                self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                self.socket.connect((self.manager_address, int(self.manager_port)))
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py:935]: I think that the `send` is missing here

File: docs/tests/integration/test_remoted/test_agent_communication/test_request_agent_info.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ state files using the remoted socket.
 
 |Tier | Number of tests | Time spent |
 |:--:|:--:|:--:|
-| 0 | 1 | 12min |
+| 0 | 1 | 3min |
```
Comment (jmv74211) [inline docs/tests/integration/test_remoted/test_agent_communication/test_request_agent_info.md:14]:
```suggestion
| 0 | 1 | 3m|
```

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: docs/tests/integration/test_remoted/test_agent_communication/test_request_agent_info.md (hunk 2 above)
Comment (Pac0x73706f74686f756e64) [reply docs/tests/integration/test_remoted/test_agent_communication/test_request_agent_info.md]: done here ty https://github.com/wazuh/wazuh-qa/pull/1106/commits/c6c8bd3ac9e94d39e6045c73a0a8ca6df73a9080

File: deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py (hunk 1 above)
Comment (Pac0x73706f74686f756e64) [reply deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py:935]: Yes, you're right, thanks! https://github.com/wazuh/wazuh-qa/pull/1106/commits/c6c8bd3ac9e94d39e6045c73a0a8ca6df73a9080

File: deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py [hunk 3]
```diff
@@ -84,26 +90,29 @@ class Agent:
...
         stop_receive (int): Flag to determine when to activate and deactivate the agent event listener.
         stage_disconnect (str): WPK process state variable.
-        debug (boolean): enable debug logging level.
+        rcv_msg_limit (int): max elements for the received message queue.
+        rcv_msg_queue (deque): Doubly Ended Queue to store received messages in the agent.
     """
     def __init__(self, manager_address, cypher="aes", os=None, inventory_sample=None, rootcheck_sample=None,
                  id=None, name=None, key=None, version="v3.12.0", fim_eps=None, fim_integrity_eps=None,
-                 authd_password=None, debug=True, disable_all_modules=False):
+                 authd_password=None, debug=True, disable_all_modules=False, rcv_msg_limit=100):
```
[the remark this answers is not in the document]
Comment (Pac0x73706f74686f756e64) [reply deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py]: Sure, I've removed the debug option, actually. https://github.com/wazuh/wazuh-qa/pull/1106/commits/c6c8bd3ac9e94d39e6045c73a0a8ca6df73a9080

File: deps/wazuh_testing/wazuh_testing/remote.py [hunk 4]
```diff
@@ -97,7 +95,8 @@ def callback_error_invalid_ip(ip):
 
 
 def callback_info_no_allowed_ips():
-    msg = fr"INFO: \(\d+\): IP or network must be present in syslog access list \(allowed-ips\). Syslog server disabled."
+    msg = fr'INFO: \(\d+\): IP or network must be present \
+    in syslog access list \(allowed-ips\). Syslog server disabled.'
```
[the remark this answers is not in the document]
Comment (Pac0x73706f74686f756e64) [reply deps/wazuh_testing/wazuh_testing/remote.py]: sure https://github.com/wazuh/wazuh-qa/pull/1106/commits/c6c8bd3ac9e94d39e6045c73a0a8ca6df73a9080

File: deps/wazuh_testing/wazuh_testing/remote.py [hunk 5]
```diff
@@ -62,7 +59,8 @@ def callback_error_getting_protocol():
 
 
 def callback_warning_syslog_tcp_udp():
-    msg = fr"WARNING: \(\d+\): Only secure connection supports TCP and UDP at the same time. Default value \(TCP\) will be used."
+    msg = fr"WARNING: \(\d+\): Only secure connection supports TCP and UDP at the same time. Default value \(TCP\) \
+    will be used. "
```
[the remark this answers is not in the document]
Comment (Pac0x73706f74686f756e64) [reply deps/wazuh_testing/wazuh_testing/remote.py]: sure https://github.com/wazuh/wazuh-qa/pull/1106/commits/c6c8bd3ac9e94d39e6045c73a0a8ca6df73a9080

**Review by jmv74211 — APPROVED** [review]

**Review by BraulioV — APPROVED** [review]

**Pull request merged**